frontend/src/component/admin/billing/BillingInvoices/BillingInvoice/BillingInvoiceUsageRow/BillingInvoiceUsageRow.tsx

```diff
@@ -17,43 +17,64 @@ const StyledCellWithIndicator = styled('div')(({ theme }) => ({
 
 type BillingInvoiceUsageRowProps = DetailedInvoicesLineSchema & {
     invoiceCurrency?: string;
+    invoiceStatus?: string;
 };
 
 export const BillingInvoiceUsageRow = ({
     quantity,
     consumption,
     limit,
     description,
-    currency,
     totalAmount,
+    unitPrice,
     invoiceCurrency,
+    invoiceStatus,
 }: BillingInvoiceUsageRowProps) => {
     const percentage =
         limit && limit > 0
             ? Math.min(100, Math.round(((consumption || 0) / limit) * 100))
             : undefined;
 
-    const hasAmount = totalAmount && totalAmount > 0;
+    const isEstimate = invoiceStatus === 'estimate';
+    const overage =
+        isEstimate && consumption && limit
+            ? Math.max(0, consumption - limit)
+            : quantity;
+    const includedAmount =
+        isEstimate && consumption && limit
+            ? Math.min(consumption, limit)
+            : consumption;
+    const calculatedAmount =
+        isEstimate && unitPrice && consumption && limit
+            ? Math.max(0, consumption - limit) * unitPrice
+            : totalAmount;
```

Comment:
`consumption && limit` part appears in many places. Does it make sense to name it, or have a code path that wraps it?